src/git.rs

```diff
@@ -7,22 +8,38 @@ pub fn get_root_directory_path<F>(
     target_directory: Option<&str>,
 ) -> Result<Option<String>, Option<String>>
 where
-    F: Fn(&str, Option<&str>, bool) -> Result<Option<String>, Option<String>>,
+    F: Fn(
+        &str,
+        Option<&str>,
+        bool,
+        Option<&HashMap<String, String>>,
+    ) -> Result<Option<String>, Option<String>>,
```

Comment:
Once trait aliases land on stable we should take advantage of them and simplify these signatures